lib/vagrant-saltdeps/provisioner.rb

```diff
@@ -40,8 +40,7 @@ def checkout_deps
           else
             g = Git.clone(uri, name, path: @checkout_path)
           end
-          g.branch(branch).checkout
-          g.fetch
+          g.checkout(branch)
```

Comment:
I think we should still do a  g.fetch here to pull the latest code if we are already on a branch.